src/ngraph/op/roi_align.cpp

```diff
@@ -0,0 +1,193 @@
+    , m_spatial_scale{spatial_scale}
+    , m_mode{mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v3::ROIAlign::validate_and_infer_types()
+{
+    NODE_VALIDATION_CHECK(
+        this,
+        get_input_element_type(0).is_real() && get_input_element_type(1).is_real(),
+        "The data type for input and ROIs is expected to be a floating point type. Got: ",
+        get_input_element_type(0),
+        " and: ",
+        get_input_element_type(1));
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(2).is_integral_number(),
+                          "The data type for batch indices is expected to be an integer. Got: ",
+                          get_input_element_type(2));
+
+    const auto& input_ps = get_input_partial_shape(0);
+    NODE_VALIDATION_CHECK(this,
+                          input_ps.rank().compatible(4),
+                          "Expected a 4D tensor for the input data. Got: ",
+                          input_ps);
+
+    const auto& rois_ps = get_input_partial_shape(1);
+    NODE_VALIDATION_CHECK(this,
+                          rois_ps.rank().compatible(2),
+                          "Expected a 2D tensor for the ROIs input. Got: ",
+                          rois_ps);
+
+    const auto& batch_indices_ps = get_input_partial_shape(2);
+    NODE_VALIDATION_CHECK(this,
+                          batch_indices_ps.rank().compatible(1),
+                          "Expected a 1D tensor for the batch indices input. Got: ",
+                          batch_indices_ps);
+
+    if (rois_ps.rank().is_static())
+    {
+        const auto rois_second_dim = rois_ps[1];
+        NODE_VALIDATION_CHECK(this,
+                              rois_second_dim.compatible(4),
+                              "The second dimension of ROIs input should contain box coordinates. ",
+                              "This dimension is expected to be equal to 4. Got: ",
+                              rois_second_dim);
+
+        if (batch_indices_ps.is_static())
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                rois_ps[0].same_scheme(batch_indices_ps[0]),
+                "The first dimension of ROIs input must be equal to the first dimension ",
+                "of the batch indices input. Got: ",
+                rois_ps[0],
+                " and: ",
+                batch_indices_ps[0]);
+        }
+    }
+
+    // the output shape should have the following format [NUM_ROIS, C, pooled_h, pooled_w]
+    auto output_shape = PartialShape{{Dimension::dynamic(),
+                                      input_ps[1],
+                                      Dimension{static_cast<int64_t>(m_pooled_h)},
+                                      Dimension{static_cast<int64_t>(m_pooled_w)}}};
+
+    // if either of those 2 dimensions is static its value will be used
+    // for the first dimension of the output shape - 'NUM_ROIS'
+    if (rois_ps[0].is_static())
```

Comment:
I get another segfault here.
gdb check:
 `rois_ps = (const ngraph::PartialShape &){m_rank_is_static = false, m_dimensions = std::vector of length 0, capacity 0}`